expan/core/experiment.py

```diff
@@ -287,11 +288,70 @@ def _quantile_filtering(self, data, kpis, percentile, threshold_type):
         :return: boolean values indicating whether the row should be filtered
         :rtype: pd.Series
         """
-        method_table = {'upper': lambda x: x > threshold, 'lower': lambda x: x <= threshold}
-        na_replacement={'upper': -float_info.max        , 'lower':  float_info.max}
-
+        method_table   = {'upper': lambda x: x > threshold, 'lower': lambda x: x <= threshold}
+        na_replacement = {'upper': -float_info.max,         'lower':  float_info.max}
 
         for column in data[kpis].columns:
             threshold = np.percentile(data[column].fillna(na_replacement[threshold_type]), percentile)
             flags = flags | data[column].apply(method_table[threshold_type])
         return flags
+
+
+    def variant_split_is_consistent_with_expected_distribution(self, variant_column, weights, min_counts=5, alpha=0.05):
+        """ Tests the consistency of variant split with the hypothesized distribution.
+        
+        :param variant_column: variant column from the input data frame
+        :param weights: list of dicts with weights/split percentage per each variant 
+                        ([{'value':<variant>,'weight':<weight>}, ...])
+        :param min_counts: minimum number of observed and expected frequencies (should be at least 5), see 
+                            http://docs.scipy.org/doc/scipy-0.16.1/reference/generated/scipy.stats.chisquare.html
+        :param alpha: significance level, 0.05 by default
+        :return: True(if split is consistent with the given split) or
+                 False(if split is not consistent with the given split)
+        :rtype:  Boolean, float, float
+        """
+        if not variant_column:
+            logger.warning("Variant split check was cancelled since input variant column is empty or doesn't exist.")
+            return None
```

Comment:
What about raising an exception if calling code doesn't fulfil method's contract? 
It will help to simplify caller code, you won't have to worry about special handling of Nones